### PR TITLE
Fix Enum projection bug

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
This PR fixes a bug where `GroupPopulation.value_nth_person(EnumArray) -> Array[int]`, instead of the expected `GroupPopulation.value_nth_person(EnumArray) -> EnumArray`. I've extended the previous test case to cover this as well, to avoid code duplication.

### CODE REVIEW

**Comment by nikhilwoodruff** [comment]
@benjello I added a test to cover this, should I add more? I think it covers the only use case.

**Review by benjello — APPROVED** [review]

**Comment by benjello** [comment]
@MattiSG @maukoquiroga we need an additional review here ;-)

**Comment by nikhilwoodruff** [comment]
Hey @sandcha @MattiSG @maukoquiroga - if you've got a spare second and this looks OK to add, would be much appreciated as we're having to use some workarounds in the US system (string type variable copies of affected Enums) that end up causing their own bugs. Thanks in advance!

File: openfisca_core/populations/group_population.py [hunk 1]
```diff
@@ -267,6 +267,10 @@ def value_nth_person(self, n, array, default = 0):
         # The map is needed b/c the order of the nth persons of each household in the persons vector is not necessarily the same than the household order.
         result[nb_persons_per_entity > n] = array[members_map][positions[members_map] == n]
 
+        # Preserve Enum dtype
+        if isinstance(array, EnumArray):
+            result = EnumArray(result, array.possible_values)
```
Comment (sandcha) [inline openfisca_core/populations/group_population.py:272]: Thank you for this fix that also brings some consistency with the `value_from_person(...)` treatment for enums.

File: openfisca_core/populations/group_population.py [hunk 2]
```diff
@@ -267,6 +267,10 @@ def value_nth_person(self, n, array, default = 0):
         # The map is needed b/c the order of the nth persons of each household in the persons vector is not necessarily the same than the household order.
         result[nb_persons_per_entity > n] = array[members_map][positions[members_map] == n]
 
+        # Preserve Enum dtype
```
[the remark this answers is not in the document]
Comment (nikhilwoodruff) [reply openfisca_core/populations/group_population.py:270]: Oh yes, you're right. Thanks.

File: tests/core/test_projectors.py [hunk 3]
```diff
@@ -287,7 +287,19 @@ class projected_family_level_variable(Variable):
         def formula(family, period):
             return family.household("household_level_variable", period)
 
-    system.add_variables(household_level_variable, projected_family_level_variable)
+    class decoded_projected_family_level_variable(Variable):
```
Comment (augusto-herrmann) [inline tests/core/test_projectors.py:290]: I wonder why the class is named like this, when Python class naming conventions is to use [PascalCase](https://namingconvention.org/python/class-naming.html). But then again, I'm not familiar enough with the code of this project so I'm sure there is a perfectly valid reason for this.

**Comment by nikhilwoodruff** [comment]
> @nikhilwoodruff From a "logistical" point of view, git rebase is favored in openfisca-core in order to get a linear history (easier to read etc.). Could you use git rebase instead of git merge?
> 
> > You can for example compare their impacts by looking into the branch history with `git log --pretty=format:'%h - %an : %s' --date-order --graph`.

Thanks @sandcha - I've updated the version number and changelog, so all the checks now pass. As for the rebasing, sounds good; I don't think I have permissions to merge the PR though?

**Comment by nikhilwoodruff** [comment]
Re-merging after a recent PR was merged to ensure the versioning is still up-to-date here.

**Comment by nikhilwoodruff** [comment]
Thanks @benjello, @sandcha and @augusto-herrmann for the help with this PR!

**Pull request merged**